cmd/snap/cmd_run.go

```diff
@@ -760,6 +760,36 @@ func snapdHelperPath(toolName string) (string, error) {
 	return filepath.Join(snapBase, dirs.CoreLibExecDir, toolName), nil
 }
 
+/* Kerberos tickets live in /tmp, or in the place specified by KRB5CCNAME
+ * environment variable. However, in snaps /tmp is private.
+ * So rewire the environment variable to point to /var/lib/snapd/hostfs/tmp,
+ * in case the variable is unset or resolves tickets to /tmp/krb5cc*.
+ *
+ * Snaps that want to read Kerberos tickets must then connect to the
+ * kerberos-tickets interface to have access to the tickets.
+ */
+func exposeKerberosTickets(info *snap.Info) (string, error) {
+	ok := false
+
+	krb5EnvVar := osGetenv("KRB5CCNAME")
+	if len(krb5EnvVar) == 0 {
+		ok = true
+	} else if strings.HasPrefix(krb5EnvVar, "FILE:") {
+		path := strings.TrimPrefix(krb5EnvVar, "FILE:")
+		path = filepath.Clean(path)
+		ok = strings.HasPrefix(path, "/tmp/krb5cc")
+	} else if strings.HasPrefix(krb5EnvVar, "DIR:/tmp") {
+		path := strings.TrimPrefix(krb5EnvVar, "DIR:")
+		path = filepath.Clean(path)
+		ok = path == "/tmp"
+	}
+
+	if ok {
+		return "/var/lib/snapd/hostfs/tmp", nil
+	}
+	return "", fmt.Errorf("Unsupported KRB5CCNAME: %s", krb5EnvVar)
```

Comment:
This is all okay but it feels very C-like

I'd write it more directly

If env is empty, return "", nil
If env has FILE: prefix, validate and return
If env has DIR: prefix, validate and return
Otherwise return "", error

I'd probably use switch/case for the matching